src/PKSim.Core/Commands/ResetParameterCommand.cs

```diff
@@ -32,18 +32,13 @@ protected override void UpdateParameter(IParameter parameter, IExecutionContext
 
       private void resetValueOriginForDefaultParameter(IParameter parameter, IExecutionContext context)
       {
-         //TODO MBD: That does not work quite yet
-         var parametersInContainerRepository = context.Resolve<IParametersInContainerRepository>();
+         var valueOriginRepository = context.Resolve<IValueOriginRepository>();
 
-         var parameterMetaData = parametersInContainerRepository.ParameterMetaDataFor(parameter);
+         var valueOrigin = valueOriginRepository.ValueOriginFor(parameter);
```

Comment:
That should take care of it now. The ValueOriginRepository is our guy having a cache of all ValueOrigin. Since we have the id saved, we can try to retrieve it again